department-of-veterans-affairs/vets-api
# [53040] Mocked Authentication `authorize` route setup

## Summary

- This PR adds the `authorize` route to the new Mocked Authentication module. The route currently takes (and validates) a CSP `type` param as a URI-encoded JSON string through the `credential_info` URL param. If the request passes this validation the JSON hash will be stored with a generated code as its key, this code is then returned as the response:
  ```
  {"credential_info_code":"13d1ffc6bb2e99897a746523fb69a1e5"}
  ```
- To test, go to the following route:
  ```
  http://localhost:3000/mocked_authentication/authorize?credential_info=%7B%22type%22:%22logingov%22%7D
  ```
  You should receive a `credential_info_code`, which you can then use in a Rails console to look up the corresponding Redis data:
  ```
  2.7.0 :002 > MockedAuthentication::MockCredentialInfo.find('13d1ffc6bb2e99897a746523fb69a1e5')
   => #<MockedAuthentication::MockCredentialInfo:0x000055594f78a7f0 @credential_info_code="13d1ffc6bb2e99897a746523fb69a1e5", @credential_info={"type"=>"logingov"}, @persisted=true, @validation_context=nil, @errors=#<ActiveModel::Errors:0x000055594f5ffae8 @base=# <MockedAuthentication::MockCredentialInfo:0x000055594f78a7f0 ...>, @errors=[]>> 
  ```

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#53040

## Testing done

- Manual testing of feature, safety authentication tests, unit specs added to cover new feature.

## Acceptance criteria


## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
